homeassistant/components/ezviz/camera.py

```diff
@@ -32,166 +34,159 @@
             )
 
         else:
-            _LOGGER.info(
-                "Found camera with serial %s without configuration. Add it to configuration.yaml to see the camera stream",
-                camera_serial,
-            )
 
-        camera["username"] = camera_username
-        camera["password"] = camera_password
-        camera["rtsp_stream"] = camera_rtsp_stream
-
-        camera["ezviz_camera"] = EzvizCamera(ezviz_client, camera_serial)
-
-        camera_entities.append(HassEzvizCamera(**camera))
-
-    add_entities(camera_entities)
-
-
-class HassEzvizCamera(Camera):
-    """An implementation of a Foscam IP camera."""
-
-    def __init__(self, **data):
-        """Initialize an Ezviz camera."""
-        super().__init__()
-
-        self._username = data["username"]
-        self._password = data["password"]
-        self._rtsp_stream = data["rtsp_stream"]
-
-        self._ezviz_camera = data["ezviz_camera"]
-        self._serial = data["serial"]
-        self._name = data["name"]
-        self._status = data["status"]
-        self._privacy = data["privacy"]
-        self._audio = data["audio"]
-        self._ir_led = data["ir_led"]
-        self._state_led = data["state_led"]
-        self._follow_move = data["follow_move"]
-        self._alarm_notify = data["alarm_notify"]
-        self._alarm_sound_mod = data["alarm_sound_mod"]
-        self._encrypted = data["encrypted"]
-        self._local_ip = data["local_ip"]
-        self._detection_sensibility = data["detection_sensibility"]
-        self._device_sub_category = data["device_sub_category"]
-        self._local_rtsp_port = data["local_rtsp_port"]
-
-        self._ffmpeg = None
-
-    def update(self):
-        """Update the camera states."""
-
-        data = self._ezviz_camera.status()
-
-        self._name = data["name"]
-        self._status = data["status"]
-        self._privacy = data["privacy"]
-        self._audio = data["audio"]
-        self._ir_led = data["ir_led"]
-        self._state_led = data["state_led"]
-        self._follow_move = data["follow_move"]
-        self._alarm_notify = data["alarm_notify"]
-        self._alarm_sound_mod = data["alarm_sound_mod"]
-        self._encrypted = data["encrypted"]
-        self._local_ip = data["local_ip"]
-        self._detection_sensibility = data["detection_sensibility"]
-        self._device_sub_category = data["device_sub_category"]
-        self._local_rtsp_port = data["local_rtsp_port"]
-
-    async def async_added_to_hass(self):
-        """Subscribe to ffmpeg and add camera to list."""
-        self._ffmpeg = self.hass.data[DATA_FFMPEG]
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": SOURCE_DISCOVERY},
+                    data={
+                        ATTR_SERIAL: camera[ATTR_SERIAL],
+                        CONF_USERNAME: None,
```

Comment:
We don't need to send items that don't have useful values.

Reply:
Done.